app/main/lib/image_similarity.py

```diff
@@ -145,3 +146,46 @@ def search_by_phash(phash, threshold, context, limit=None):
   except Exception as e:
     db.session.rollback()
     raise e
+
+
+@tenacity.retry(wait=tenacity.wait_fixed(0.5), stop=tenacity.stop_after_delay(5), after=_after_log)
+def search_by_pdq(pdq, threshold, context, limit=None):
+  try:
+    context_query, context_hash = get_context_query(context)
+    if context_query:
+        cmd = """
+          SELECT * FROM (
+            SELECT id, sha256, pdq, url, context, bit_count_image_new(pdq # :pdq)
+            AS score FROM images
+          ) f
+          WHERE score >= :threshold
+          AND 
+          """+context_query+"""
+          ORDER BY score ASC
+        """
+    else:
+        cmd = """
+          SELECT * FROM (
+            SELECT id, sha256, pdq, url, context, bit_count_image_new(pdq # :pdq)
+            AS score FROM images
+          ) f
+          WHERE score >= :threshold
+          ORDER BY score ASC
+        """
+    if limit:
+        cmd = cmd+" LIMIT :limit"
+    matches = db.session.execute(text(cmd), dict(**{
+      'pdq': pdq,
+      'threshold': threshold,
+      'limit': limit,
+    }, **context_hash)).fetchall()
+    keys = ('id', 'sha256', 'pdq', 'url', 'context', 'score')
+    rows = []
+    for values in matches:
+      row = dict(zip(keys, values))
+      row["model"] = "image"
```

Comment:
What do you think about setting `model` equal to `image/pdq` to explicitly differentiate it from the previous phash results @DGaffney @ahmednasserswe ?

Reply:
That's a cheap thing to do that may disentangle a ton of issues to come in the future. I think we should do it.